### ISSUE TITLE
Failing unit tests

### ISSUE THREAD

**Issue by jvermillard** [eclipse/wakaama#212]
With current master I have the following unit test failure:
```
    1. /home/jvermillar/sandbox/wakaama/tests/tlv_json_lwm2m_data_test.c:113  - CU_ASSERT_EQUAL(original_length,length)
    2. /home/jvermillar/sandbox/wakaama/tests/tlv_json_lwm2m_data_test.c:122  - CU_FALSE
```

**Comment by dnav** [comment]
JSON unit tests need improvements. The returned buffer is compared to a hard-coded one. It should be parsed instead.

**Comment by davidgraeff** [comment]
I disagree. Hardcoded is suitable here, as there is no 3rd party json library pulled in and everything is in a controlled condition. But the unit tests need to be adapted if the core is changed, of course.

**Comment by dnav** [comment]
See #324 
JSON tests 9 and 10 are still failing.
9 because it contains Object Links which are not handled by the JSON encoder.
10 because a float value exceeds the encoder limits.

**Comment by dnav** [comment]
Fixed by #324
